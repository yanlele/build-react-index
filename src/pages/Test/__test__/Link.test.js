/*快照测试*/
import React from 'react';
import {render} from 'enzyme';
import toJson from 'enzyme-to-json';
import ALink from '../Link';

describe('Link', () => {
  it('正确的快照', function () {
    const tree = render(<ALink page="http://www.facebook.com">facebook</ALink>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('这个是正确的快照更改', function () {
    const tree = render(<ALink page="http://www.yanleweb.com">yanlelele</ALink>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('这种时候，快照就要报错了', function () {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20),
      name: 'yanle',
    };
    expect(user).toMatchSnapshot({
      createAt: expect.any(Date),
      id: expect.any(Number),
    });
  });
});
