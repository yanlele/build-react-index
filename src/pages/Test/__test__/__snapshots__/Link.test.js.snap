// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link 正确的快照 1`] = `
<a
  class="normal"
  href="http://www.facebook.com"
>
  facebook
</a>
`;

exports[`Link 这个是正确的快照更改 1`] = `
<a
  class="normal"
  href="http://www.yanleweb.com"
>
  yanlelele
</a>
`;

exports[`Link 这种时候，快照就要报错了 1`] = `
Object {
  "createAt": Any<Date>,
  "id": Any<Number>,
  "name": "yanle",
}
`;
